TGS3 DM changes

Cyberboss opened the pull request:
Might as well throw this out there so yall can yell at me.

Moved to a service oriented architecture for the meta server. It now runs as a windows service and can be controlled via external programs.

To compile: You'll need a version of VS that can compile .NET 4.5 and the WiX framework
If you wanna test it out you can just download my latest WIP here (built by CI): https://github.com/Cyberboss/tgstation/releases/latest

Has all the bells and whistles of the old server and more. Readme: https://github.com/Cyberboss/tgstation/blob/tgs3/tools/tgstation-server-3/README.md

TODO:
- [x] BYOND version management and auto updates
- [x] Detailed logging with windows events (e.g. revert to last good revision for when I break the server)
- [x] PRs listed with the SHA they were testmerged with
- [x] Built in IRC and Discord support. And easy to add more providers
- [x] True hard restarts from DS
- [x] Game configuration GUI
- [x] A framework for passing messages to and from the world
- [x] Server Contol Panel (Repo/BYOND/Server pages done)
- [x] Automatic login and startup
- [x] CLI support
- [x] Now handles all IRC commands
- [x] Get changelog generation working
- [x] Add a cancel button for compilation
- [x] Add a merge pr CL command
- [x] Have the service download testmerge JSON

Also enabled appveyor auto building and releasing so we don't get msi trust issues.
Example Build: https://ci.appveyor.com/project/Cyberboss/tgstation/build/1.0.341
Releases are only created when a commit is created with `[TGSDeploy]` in the title

DM code changelog:

- Removed HttpsGet and revdata PR detail downloading. It's handled by the service now
- New config option to IRC announce new rounds
- New 'disabled' field for map configuration, used to disable a map without commenting it out
- Gives admins the option of the hardest possible reboot by telling the service to kill and restart DD
- Improved IRC ahelps

More below

:cl:
experiment: New server backend!
tweak: Test merged PRs now show the commit of the PR at which they were merged
/:cl:

More TODOs:

- [ ] Allow the service to login to the pager
- [ ] Add optional segmenting of the DD process into it's own user account
- [ ] Add options to the installer

MrPerson commented:
Nice commits

optimumtact commented:
Who makes a management tool with no CLI, this is so windows it physically hurts.

Cyberboss commented:
The CLI is the easiest part to do now that the back end is done

On Apr 24, 2017 7:04 PM, "oranges" <notifications@github.com> wrote:

> Who makes a management tool with no CLI, this is so windows it physically
> hurts.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/26534#issuecomment-296846112>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHyweo_yeUDU83OXBKsx2sMlEd2hGAMDks5rzSqQgaJpZM4NGzw9>
> .
>

optimumtact commented:
It'S HURTING ME INSIDE

optimumtact commented on code/controllers/subsystem/garbage.dm:
```diff
@@ -175,12 +175,13 @@ SUBSYSTEM_DEF(garbage)
 /proc/qdel(datum/D, force=FALSE)
 	if(!D)
 		return
+	if(!istype(D))
```
What be this

optimumtact replied:
@MrStonedOne how would a thing not be a datum?

MrStonedOne replied:
it could be a number, a string, a matrix, a file object.

optimumtact commented:
I'm against this because it removes the old python bot now :-1:

Cyberboss commented:
The nostalgia trench is real

MrStonedOne commented on bot/minibot.py:
```diff
@@ -1,162 +0,0 @@
-#!/usr/bin/env python3
-# This bot was made by tkdrg.
-# Ask #coderbus@irc.rizon.net if this breaks.
-# See LICENSE-bot_folder.txt for the license of the files in this folder.
```
Deleting minibot is premature.

if your system becomes the primary system sure, we can discuss it then it (would still be likely rejected by the maintainers since minibot is more portable then your system is), but until then it might be best to keep this around and keep the game compatible with both

optimumtact commented:
it's not nostalgia but MSO covered all the reasons more succinctly so I'll leave it at that.

MrStonedOne commented:
I just noticed the vs 2017 requirement to dev on.

Could you explain in detail why it needs a high requirement?

It seems it's mostly the installer framework, but they make that going down to 2008 still maintained.

Cyberboss commented:
Yeah, the dev environment isn't a requirement as long as the c# can compile. The installer is the only thing that requires 2017 basically. As long whatever you use can compile .NET 4.5 it should be fine

Cyberboss commented:
As an aside, you can install the service manually by running
```
C:\WINDOWS\Microsoft .NET\Framework\v4.<some version string>\InstallUtil.exe <path to service executable>
```

MrStonedOne commented:
ya, if you don't, i will at some point change the install framework to something that supports  vs2012

Cyberboss commented:
Whats the highest version 2012 supports?

Iamgoofball commented on tools/appveyor/build-tgs3-installer.cmd:
```diff
@@ -0,0 +1,2 @@
+set DEVENV="C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\devenv.exe"
```
smells like alternate install location support is nonexistant

Cyberboss replied:
CI SERVICE

Iamgoofball commented on tools/appveyor/vs-2017-installer-templates.ps1:
```diff
@@ -0,0 +1,3 @@
+$vsixPath = "$($env:USERPROFILE)\InstallerProjects.vsix"
+(New-Object Net.WebClient).DownloadFile('https://visualstudiogallery.msdn.microsoft.com/fd136a01-a0c8-475f-94dd-240136f86746/file/247375/3/InstallerProjects.vsix', $vsixPath)
+& "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VSIXInstaller.exe" /q /a $vsixPath
```
smells like alternate install location support is nonexistant

Cyberboss replied:
CI SERVICE

Iamgoofball commented on tools/tgstation-server-3/ControlPanel/App.config:
```diff
@@ -0,0 +1,45 @@
+                <value>False</value>
+            </setting>
+            <setting name="ByondPath" serializeAs="String">
+                <value>C:\Program Files (x86)\BYOND\bin</value>
```
smells like alternate install location support is nonexistant

Cyberboss replied:
CONFIG FILE!!!

Iamgoofball commented on tools/tgstation-server-3/ControlPanel/TGControlPanel.csproj:
```diff
@@ -0,0 +1,118 @@
+    </Reference>
+    <Reference Include="System.Data" />
+    <Reference Include="System.Deployment">
+      <HintPath>C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.Deployment.dll</HintPath>
```
smells like alternate install location support is nonexistant

8 of them

Cyberboss replied:
...

Iamgoofball commented on tools/tgstation-server-3/ServerService/App.config:
```diff
@@ -0,0 +1,125 @@
+                <value>False</value>
+            </setting>
+            <setting name="ByondPath" serializeAs="String">
+                <value>C:\Program Files (x86)\BYOND\bin</value>
```
smells like alternate install location support is nonexistant

Cyberboss replied:
Default value

Iamgoofball replied:
No exceptions

Iamgoofball reviewed (changes requested):
smells like alternate install location support is nonexistant

Cyberboss commented:
Goof, do you not know what appveyor is?

Cyberboss commented:
@MrStonedOne I changed the installer framework to WiX: http://wixtoolset.org/

Also enabled appveyor auto building and releasing so we don't get msi trust issues.

Example Build: https://ci.appveyor.com/project/Cyberboss/tgstation/build/1.0.113

And release: https://github.com/Cyberboss/tgstation/releases/tag/tgstation-server-v3.0.61

Releases are only created when a tag is pushed

Iamgoofball commented on appveyor.yml:
```diff
@@ -15,14 +15,13 @@ build:
   provider: GitHub
   auth_token:
-    secure: github_oath_token
+    secure: 9Yf+FtoE7nf36V3dyVRANkNZ7og4dp/0uojCS51vrvIeNOTIM8Nq24lylgVBOhSL
```
smells like not a config

MrStonedOne replied:
Smells like a leaked api key.

Cyberboss replied:
It's an encrypted thing

MrStonedOne replied:
Can't be and still be useful.

Encrypted how?

Decrypted how?

What is it used for?

At the end of the day either somebody could still abuse this and auth themselves with the token, or the token is useless.

MrStonedOne replied:
also, what the fuck is appveyor?

Cyberboss replied:
Appveyor is windows CI. The token is a repo scope github api token that's encrypted using a key stored in your appveyor account and decrypted on the build box. It only is given to builds that aren't from pull requests. And it's used for automatically pushing to github releases like so: https://github.com/Cyberboss/tgstation/releases/latest

Cyberboss replied:
Released from here: https://ci.appveyor.com/project/Cyberboss/tgstation

optimumtact commented on appveyor.yml:
```diff
@@ -20,7 +20,7 @@ deploy:
   description: 'The /tg/station server suite'
   provider: GitHub
   auth_token:
-    secure: 9Yf+FtoE7nf36V3dyVRANkNZ7og4dp/0uojCS51vrvIeNOTIM8Nq24lylgVBOhSL
```
I saw it

Cyberboss replied:
its an encrypted thing so w/e

Cyberboss replied:
in fact, i'm certain it HAS to be pasted in

Cyberboss commented:
Changelogs working https://github.com/Cyberboss/tgstation/commit/46a7f1d4bab917d965f5c762d88f047fcd5489a3

Iamgoofball commented:
Cloud support for Windows XP?

Cyberboss commented:
.NET 4.5, so no sadly. Get 7

Iamgoofball commented:
# muh virtualbox ban evading wtf i hate cyberboss now???

Iamgoofball commented:
# i swear i use virtualbox inside wine to run windows xp with the cid spoofer because of compatability its not like im ban evading or anything

Cyberboss commented:
Unless you're hosting a server, why would this even matter?

Iamgoofball commented:
# muh ~~ban evasion~~windows xp

optimumtact commented on html/changelog.html:
```diff
@@ -1309,23 +1309,6 @@ <h3 class="author">coiax updated:</h3>
 				<li class="bugfix">Cyborg radios can no longer have their inaccessible wires pulsed by EMPs.</li>
 			</ul>
-
-			<h2 class="date">03 March 2017</h2>
```
was this necessary to remove?

optimumtact replied:
nobody should be touching this file but the changelog bot

Cyberboss replied:
Yeah, that was just me testing

Iamgoofball commented:
so real talk how do I do this

Cyberboss commented:
@Iamgoofball Download the latest revision here: https://github.com/Cyberboss/tgstation/releases/latest. Extract the files and run the installer. It'll put a shortcut to the (unfinished) control panel on your desktop and add the (mostly finished) TGCommandLine.exe to your path. You can run `TGCommandLine.exe ?` for info on commands

Some things you can do from there for testing (I wrote them in commandline form, but all these steps except 1, 2, and 5 can be done with the control panel):

- Set the install directory (Default is C:\tgstation-server-3 `TGCommandLine.exe config move-server <path you want>`)
- Enable the IRC bot (`TGCommandLine.exe irc enable` `TGCommandLine.exe irc nick <nickname>`)
- Clone a repo, use my branch because it has DM modifications for interfacing (`TGCommandLine.exe repo setup https://github.com/Cyberboss/tgstation.git tgs3`)
- Install the BYOND version you'll use (`TGCommandLine.exe byond update 511 1381`)
- Modify your config.txts to your liking (No interface for this yet, They are stored in C:\tgstation-server-3\Static\Config by default)
- Once the repo is done cloning (You can view progress in the control panel). Set up the game folder `TGCommandLine.exe dm initialize`
- Once that's done, you can run `TGCommandLine.exe dm compile` to compile it
- Finally run `TGCommandLine.exe dd start` to start the server. It'll run on byond://127.0.0.1:2337 by default

Cyberboss commented:
And you can always remove it all from Add/Remove programs

Iamgoofball commented:
can this testmerge?

Iamgoofball commented:
like, can i set it up locally and access testmerge stuff

Cyberboss commented:
Command line is done

Cyberboss commented:
@Iamgoofball Yeah, test merges work fine

Iamgoofball commented:
sick does it have a gui interface made in visual basic?

Cyberboss commented:
Not visual basic, but close enough. Gimme a sec, I'll post some screens

Cyberboss commented:
![untitled](https://cloud.githubusercontent.com/assets/8171642/25819978/adb9fa2e-33fd-11e7-920b-8bfff7e77176.png)

![untitled](https://cloud.githubusercontent.com/assets/8171642/25820019/d15310a6-33fd-11e7-8292-01ebc7c13410.png)

![untitled](https://cloud.githubusercontent.com/assets/8171642/25820047/ee437c8c-33fd-11e7-9564-3a4e10bb5fc5.png)

This one is WIP:

![untitled](https://cloud.githubusercontent.com/assets/8171642/25820157/548ed9b4-33fe-11e7-8946-1162239bbc81.png)

Cyberboss commented:
Config GUI isn't a restriction, there's still a way to edit the .txt's manually

MrStonedOne commented:
HOLY FUCKING COMMITS

like god damn it just stopped counting.

Cyberboss committed:
TGS3

Cyberboss commented:
@MrStonedOne fixed

Cyberboss commented on tools/tgstation-server-3/TGServerService/Interop.cs:
```diff
@@ -0,0 +1,202 @@
+		}
+
+		//Fuckery to diddle byond with the right packet to accept our girth
+		string SendTopic(string topicdata, ushort port, bool retry = false)
```
@optimumtact pls revievv if you know how topic packets work

MrStonedOne replied:
The byond packet structure is as follows:

2 bytes sequence id(only after first packet, not used here), 2 bytes command id, 2 bytes content length, content.

Your system only encodes the second length byte, so you would be limited to 255 characters. i think most versions of this system has this same flaw, but otherwise it should be fine.

The 5 empty bytes at the front are flags and id'ing the world that sent the topic, not used in our system (there is also a way to send a file using this, but I don't understand it, and there is an encrypted version that has persistent connection support, but goonstation is keeping the documentation of that that they figured out secret so they can use it to lock support of goonhub into their control.).

This is a complete (and slightly more proper) packet in php:

```php
$packet = "\x00\x83" . pack('n', strlen($str) + 6) . "\x00\x00\x00\x00\x00" . $str . "\x00";
```

MrStonedOne replied:
As for the reply, I don't actually know or understand the format for that, but here is a the export() that powers the game banners:

https://github.com/tgstation/tgstation13.org/blob/master/getserverdata.php#L10 its pretty well commented.

The reply can be a list, a string, or a number, if its a list, it may contain numbers or strings and can be associative.

Your system doesn't seem to be taking this nuance into account.

Cyberboss committed:
Fix line endings

Cyberboss commented:
Reference https://github.com/Cyberboss/tgstation/pull/3 for commits that led up to 5671494

Cyberboss committed:
Enable 16-bit long topics

JamieH commented:
i'm gonna PR your PR

Cyberboss committed:
Cleans up topic socket usage

Cyberboss committed:
Reduces and refines the IRC status throttle

Cyberboss committed:
Increase the CP status delay to a reasonable amount

Cyberboss committed:
Fixes the testmerge command not having a keyword

Cyberboss committed:
Clean up rebooting a bit

Cyberboss committed:
Get error codes from windows when symlinking fails

Cyberboss committed:
Clean up world announces

Cyberboss committed:
Aborting compilation will kill the DD process

Cyberboss committed:
Add support for changing the project name

Cyberboss commented:
K

On May 10, 2017 3:18 AM, "JamieH" <notifications@github.com> wrote:

> i'm gonna PR your PR
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/26534#issuecomment-300397045>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHywes5ab2S1HonsayzmwCGCGN36B5Zcks5r4WTHgaJpZM4NGzw9>
> .
>

Cyberboss committed:
Removes the log page

Cyberboss committed:
Add support for compile cancellation

Cyberboss committed:
Version bumps and docs

Cyberboss committed:
Add merge-pr repo CL command

Cyberboss committed:
Fixes DM cancel command's help message

Cyberboss committed:
Refactor command line to show better formatted help text

Cyberboss committed:
Corrects a typo

Cyberboss committed:
Multi-key game options must be manually edited

Cyberboss committed:
Moving of the server installation from the control panel

Cyberboss committed:
Fix a bug with server moving

Cyberboss committed:
Corrects webclient disposal syntax

Cyberboss committed:
Service now handles the PR Json the game uses properly

Cyberboss committed:
PR listing command for CL

Cyberboss committed:
Fixes reversed testmerge and update help entries

Cyberboss committed:
Windows scheduling to help avoid reboot crashes

Cyberboss committed:
Generalization of chat infastructure

Cyberboss committed:
Brings a file name in line with everything else

Cyberboss committed:
Shutdown exceptions no longer keep the service online

Cyberboss committed:
Merge upstream

Cyberboss committed:
Enable provider switching on the backend. More thread safety

Cyberboss committed:
Support for switching, password encryption and defaults.

Cyberboss committed:
Removes boilerplate on log writing

Cyberboss committed:
Discord integration

Cyberboss committed:
Update the installer dependencies

Cyberboss committed:
Version bump

Cyberboss committed:
Adds support for getting the latest byond version

Cyberboss committed:
Fixes issue with not being able to set discord channels

Cyberboss committed:
Fix being able to reconnect if chat is disabled

Cyberboss committed:
Extra validation for interface types

Cyberboss committed:
Add the Chat page for the control panel

Cyberboss committed:
Various cleanup

Cyberboss committed:
Set read ACL on the data directory

Cyberboss commented:
![untitled](https://cloud.githubusercontent.com/assets/8171642/26069314/6edd4272-396e-11e7-9a5c-e3df1606dabb.png)

Cyberboss committed:
Remove redundant namespace usage

Cyberboss committed:
Fixes some buttons not updating the server page

Cyberboss committed:
Future proof against upcoming removal of repo data directory

Cyberboss committed:
Normalize Main declaration

Iamgoofball commented:
Make sure it replies to CTCP VERSION requests @Cyberboss so you don't get the bot banned by accident from Rizon.

Cyberboss committed:
Update the IRC library

Cyberboss committed:
Enables CTCP

Cyberboss committed:
Removes useless hack

Cyberboss committed:
Logging + enable IRC private messages

Cyberboss commented:
![untitled](https://cloud.githubusercontent.com/assets/8171642/26075271/a6954822-3982-11e7-86ad-65df8757f7a6.png)

Cyberboss committed:
Jobs config

Cyberboss committed:
And finally the maps config

Cyberboss committed:
Save the last config panel visited

Cyberboss commented:
![untitled](https://cloud.githubusercontent.com/assets/8171642/26078810/b0f87300-398e-11e7-9ff6-ee1e039f559f.png)

Cyberboss committed:
Not gonna use these

Cyberboss committed:
Minor formatting cleanup

Cyberboss committed:
Fixes the chat page not refereshing after clicking reconnect

Cyberboss committed:
Fixes server page not initializing correctly

Cyberboss committed:
Repo now defaults to tgstation github when not found

Cyberboss committed:
Revert "Set read ACL on the data directory"

This reverts commit 15b0021ec51532bca14690a884caa81e811fbc46.

Cyberboss committed:
Design the admin config page

Cyberboss committed:
Prep format the repo's admin_ranks.txt

Cyberboss committed:
Add a negative permissions field

Cyberboss committed:
IRC now RFC quits before disconnecting

Cyberboss committed:
Turns out that fixed the disconnect lag

Cyberboss committed:
Updates the admin ranks config api to work for us

Cyberboss committed:
Done with this config shit

Cyberboss commented:
Really like this one

![untitled](https://cloud.githubusercontent.com/assets/8171642/26179635/dc949b4c-3b32-11e7-8463-5528b55cde45.png)

Cyberboss committed:
@optimumtact

Cyberboss committed:
Fix this

Cyberboss committed:
Fix the .wxs

Cyberboss committed:
Try to get md5/sha1 working.

Cyberboss committed:
Add FCIV to appveyor

Cyberboss committed:
Merge upstream

Cyberboss committed:
Generalize the command class

Cyberboss committed:
Revert "Generalize the command class"

This reverts commit 5c61f6df58d66f0fea4170c8aee0cd5beaa99b5d.

Cyberboss committed:
ITS THE FUCKING SEX NUMBER!!!!

Cyberboss committed:
Final touches

Cyberboss committed:
No THESE are the final touches

Cyberboss committed:
Do not advertise

Cyberboss committed:
Revert "Do not advertise"

This reverts commit f64281d486f9ca27e39f19635ab4deacb2d7e1ac.

Cyberboss committed:
Hopefully the last version bump for long time

Cyberboss committed:
Fix line endings

Cyberboss commented:
@MrStonedOne It's done https://github.com/Cyberboss/tgstation/releases/tag/tgstation-server-v3.0.70.0

Cyberboss commented:
If you decide to test this, it's best to keep this PR test merged under the server while you do.

Cyberboss committed:
Fix default dbconfig.txt

Cyberboss commented:
In case it wasn't clear, these 4 buttons are mirrors of the update .bat files. The commands update and testmerge in the command line do the same

![untitled](https://cloud.githubusercontent.com/assets/8171642/26223534/7af1e68c-3bec-11e7-8088-fae31b143a95.png)

Cyberboss committed:
Fix Discord not checking the right admin channel

Cyberboss committed:
Fix discord listening on ALL channels instead of configured ones

Cyberboss committed:
Package the discord fixes for @JamieH

Cyberboss committed:
Format the testmerge data a little better

Cyberboss committed:
Apply 7 character clamping of commit strings

Cyberboss committed:
Fold admin hard reboot into regular reboot list

Cyberboss committed:
Backward ahelp compatibility with the adminbus bot

Cyberboss committed:
Removes an unecessary semicolon

Cyberboss committed:
Fix stray merge conflict in the config

Cyberboss committed:
Fix Newtonsoft being included by the commandline

Cyberboss committed:
Improve byond update logging

Cyberboss committed:
Chat cleanup

Cyberboss committed:
Fixes some setup non-errors from being displayed

Cyberboss committed:
Repository no longer counts being busy as being valid

Cyberboss committed:
Repository no longer valid while cloning

Cyberboss committed:
Fixes a nudge socket change issue

Cyberboss committed:
Frontend cleanup

Cyberboss committed:
Fixes CanStart race condition

Cyberboss committed:
Fixes compile cancel delays

Cyberboss committed:
Various fixes

Cyberboss committed:
More fixes

Cyberboss committed:
Better readme

Cyberboss commented:
Readme here: https://github.com/Cyberboss/tgstation/blob/tgs3/tools/tgstation-server-3/README.md

Blatantly plagiarized MSO's

Cyberboss committed:
More readme

Cyberboss committed:
Fix a config command description

Cyberboss committed:
Add missing repo status command

Cyberboss committed:
Never delete the backups

Cyberboss committed:
Log the compiles

Cyberboss committed:
More logging

Cyberboss committed:
Stuff

Cyberboss committed:
A thing happened, but I'm not sure what

Cyberboss committed:
Tiny

Cyberboss committed:
INB4 second squash

Cyberboss committed:
Version bump

Cyberboss commented:
Checking that @Cheridan got this right

MrStonedOne commented:
>Releases are only created when a tag is pushed

We don't push tags around here, how would this even work?

Cyberboss commented:
I've been thinking about that myself...

Cyberboss committed:
Shallow clones should speed things up

MrStonedOne commented:
you are missing reset and compile.

This clears test merges without updating, for if the master is fucked and you are testing fixes.

Technically you have it on the repo side.

And I had the server-tools create backup branches during update operations so you can undo an update, are you doing that and is there a gui to access these?

Also, shallow clones speed things up, but also slow things down: https://github.com/cocoapods/cocoapods/issues/4989#issuecomment-193772935 

And one final note, the buttons in repo tab worry me. atm the only reason having the server able to push has been able to work, is that none of the headmins are told this is the case, nor enough on how to utilize it, that in order to be able to push as a headmin, you have to already be smart enough to know not to.

Could you move those (or all) functions to the server process if they aren't, and then allow me to set some security settings for windows user accounts.

Then I could actually tweak what user accounts can do what when I move to giving each headmin their own windows login.

optimumtact commented:
Don't use a shallow clone

Cyberboss committed:
Regular clones

Cyberboss commented:
>11 minutes on cloning alone.

makes ya think

Cyberboss committed:
This is how it's set on travis

Cyberboss commented:
OK that's much better, it beats travis now

Cyberboss committed:
Merge branch 'master' into tgs3

Cyberboss committed:
Fix this dupe

Cyberboss committed:
Add backup tag support to backend and command line

Cyberboss committed:
Add some missing repo commands, fix GetHead. Fix reset on branches

Cyberboss committed:
Remove the interfaces for commit and push

Cyberboss committed:
Remove that generate changelog checkbox

MrStonedOne commented:
Did you not read the link I gave you.

Cyberboss commented:
It's using the exact two got commands that Travis does now, so I assume
it's fine

On May 21, 2017 3:52 PM, "Kyle Spier-Swenson" <notifications@github.com>
wrote:

> Did you not read the link I gave you.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/26534#issuecomment-302958873>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHywek5PnEeQAtGo70uIfpo9DaDJIBBjks5r8JXtgaJpZM4NGzw9>
> .
>

MrStonedOne commented:
Make it a config then, I do enough operation on the live repo manually that I'd like to have a full clone

Cyberboss commented:
OH I'm taking about appveyor. The server always uses a full clone

On May 21, 2017 4:29 PM, "Kyle Spier-Swenson" <notifications@github.com>
wrote:

> Make it a config then, I do enough operation on the live repo manually
> that I'd like to have a full clone
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/26534#issuecomment-302961059>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHywegBG2SJC9NzomBImN9mOAfJpK6VKks5r8J67gaJpZM4NGzw9>
> .
>

Cyberboss committed:
Yeah, that's a misunderstanding

Cyberboss committed:
Make changelog pushing a config, with no way to enable for now

Cyberboss committed:
Add Reset and Recompile option

Cyberboss committed:
Update readme

Cyberboss committed:
Repo page cleanup

Cyberboss commented:
New repo page:

![untitled](https://cloud.githubusercontent.com/assets/8171642/26287485/9a477c26-3e4a-11e7-9f91-aa4c87bd29bb.png)

Need to think about how to handle the commit/push security

Cyberboss committed:
Fixed NudgePort message possible repeating
Fixed Reset and Recompile option always being visible

Cyberboss committed:
Fixed compilation copy not overwriting files
Fixed compiler trying to unecessarily delete the whole A/B folder
Improved game folder initialization speed

MrStonedOne commented:
like i said, if you can make those actions be carried out by the server side, you could give the server side its own service account, and I could easily limit the file permissions of the repo folder or the ssh key

MrStonedOne commented:
and when pushing changelogs, do selective staging of files to commit, so changes elsewhere don't get commited and pushed

Cyberboss committed:
Selectively stage the html folder

Cyberboss committed:
Make the restriction a config

Cyberboss committed:
Switch to using LibGit2Sharp+SSH

Cyberboss committed:
WIP SSH support

Cyberboss committed:
Removes some success chat messages

Cyberboss committed:
Make repo authentication purely file based

Cyberboss committed:
Merge branch 'master' into tgs3

MrStonedOne commented:
I was a bit more fine grain:

https://github.com/tgstation/tgstation/blob/master/tools/tgstation-server/Update%20Server.bat#L52

Cyberboss committed:
Quick IRC fix

Cyberboss committed:
Should all work in theory...

Cyberboss committed:
More fine grained

Cyberboss committed:
Remove the username thing

Cyberboss committed:
Use the right default email for tgstation-server

Cyberboss committed:
Update the readme

Cyberboss committed:
That's worthy of a version bump

Cyberboss committed:
Speling

Cyberboss committed:
Makes it do as the readme says

Cyberboss committed:
Fix testmerge list not having a scrollbar

Cyberboss committed:
Trying out commit message based deployment [TGSDeploy]

Cyberboss committed:
Whoops

Cyberboss committed:
Testing

Cyberboss committed:
Better

Cyberboss committed:
Version Bump [TGSDeploy]

Cyberboss committed:
Need to set the var at parent scope [TGSDeploy]

Cyberboss committed:
Use the commit message

Cyberboss committed:
Try this [TGSDeploy]

Cyberboss committed:
Try just this

Cyberboss committed:
This maybe? [TGSDeploy]

Cyberboss committed:
>like [TGSDeploy]

Cyberboss committed:
Wildcard, bitches [TGSDeploy]

Cyberboss committed:
Saner title [TGSDeploy]

Cyberboss committed:
Readme update

Cyberboss committed:
This should loin ya

Cyberboss committed:
Fix it [TGSDeploy]

Cyberboss committed:
Readme, cleanup, and doc updates

Cyberboss commented:
New deployment system: Any commit that contains `[TGSDeploy]` on the first line will cause it to run. PRs excluded of course

Cyberboss committed:
Improve DD crash handling

Cyberboss committed:
Version bump [TGSDeploy]

Cyberboss committed:
TGS3 Config Changes

Cyberboss committed:
Line endings

Cyberboss committed:
Map config code change

Cyberboss committed:
Missed a few

Cyberboss committed:
Security and Visibility selectors for the Server page

Cyberboss committed:
Fixes OCD

Cyberboss committed:
Fax it

Cyberboss committed:
Fixes

Cyberboss committed:
Version bump [TGSDeploy]

Cyberboss committed:
Merge in master and config branch

Cyberboss committed:
eh

Cyberboss commented:
Alright @AnturK, go ahead

Cyberboss committed:
The word comment has lost it's meaning to me

Cyberboss committed:
This is a terrible name but whatever

Cyberboss committed:
Merge branch 'configUpdate' into tgs3

Cyberboss committed:
Support config changes

Cyberboss committed:
This is part of the code so it belongs with the code

Cyberboss committed:
ExportService now has a return value

Cyberboss committed:
Copying of the logs dir during compile for #27674

Cyberboss committed:
Version bump [TGSDeploy]

Cyberboss committed:
Removes some uneedful

Cyberboss committed:
Moves daemon config to BYOND folder, much safer

Cyberboss committed:
Fix a config comment translation miss

optimumtact commented:
What are we waiting on here?

Cyberboss commented:
@MrStonedOne ?

Cyberboss committed:
Merge upstream

Cyberboss committed:
Fix project settings issue

Cyberboss committed:
Fix config apply button not showing up after repo clone

Cyberboss committed:
Fix anchoring for Backup Tags: label

Cyberboss committed:
Version Bump [TGSDeploy]

MrStonedOne commented:
I think we are waiting for you to go more then 2 hours without any pushes so we know its complete and ready

Cyberboss commented:
Added more todos to OP

MrsTonedOne reviewed (changes requested):
Move the server clone location to the repo page where the actual clone button is. (or do both)

The installer needs a confirmation page so people know it installed correctly, it's too quick on the server (with it's raid 0 ssd and fastest single threaded processing of any processor on the market) that I didn't even see the progress bar until the third time I ran it.

The icons do not need to be on the desktop, they should be in the start menu and OPTIONALLY on the desktop (as in, it should be a user choice)

Install location needs to be configurable.

Location for storing various data needs to be better configurable, I can't have the logs and shit on the raid 0, they go on the raid 1. (I can fix this using symlinks myself, but it would be nice if this was configurable).

Edit:

Server shouldn't be listening for nudges if the irc bot is disabled.

Add a way to control the service if the management panel is ran from an admin account.

2sec4u commented:
I want the Desktop icons tbh

Cyberboss committed:
Nudge port only listens while server is running

Cyberboss committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into tgs3

Cyberboss committed:
Fix some instances of the control panel crashing when the service stops

Cyberboss committed:
Add start menu shortcuts

MrStonedOne commented:
\>already back up to 200 commits

optimumtact commented on tools/tgstation-server-3/TGControlPanel/Chat.cs:
```diff
@@ -0,0 +1,165 @@
+			switch (modifyingProvider)
+			{
+				case TGChatProvider.Discord:
+					var DPI = new TGDiscordSetupInfo(PI);
```
You can't hide from me

MrStonedOne replied:
!!!

AnturK commented:
Anyway, i was serious about putting this in separate repository and as sub-repo in main one. Most people won't need the server tools and we can keep appveyor only on that repository.

@tgstation/commit-access thoughts.

AnturK commented:
Yeah i meant submodule by the sub-repo, i always forget the name.

lzimann commented:
Moving server tools to a different place other than the main repo is a good idea.

Cyberboss commented:
That's fine with me

On May 31, 2017 9:49 AM, "Leo" <notifications@github.com> wrote:

> Moving server tools to a different place other than the main repo is a
> good idea.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/26534#issuecomment-305192427>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHyweqlc3uAMsJtYWKOvHAd92u_u-qQ5ks5r_XAAgaJpZM4NGzw9>
> .
>

Cyberboss committed:
Remove the actual server

Cyberboss commented:
I need to audit the changes to make sure they aren't breaking give me a bit

Cyberboss committed:
Remove appveyor

Cyberboss committed:
Fix gitignore

Cyberboss committed:
And this

Cyberboss committed:
Readd HTTPS_Get for now

Iamgoofball reviewed (commented):
safe for sdql

Cyberboss committed:
Readd legacy support

Cyberboss committed:
Fix

Cyberboss committed:
Fix this stuff

Cyberboss committed:
Last thing

Cyberboss committed:
Line endings

Cyberboss committed:
Final touches

Cyberboss committed:
Dat newline

Cyberboss committed:
More stuff

Cyberboss committed:
Where'd that go?

Cyberboss committed:
Real final touches

Cyberboss commented:
Done

AnturK commented on code/world.dm:
```diff
@@ -108,8 +112,9 @@
-		if(world.time - last_irc_status < IRC_STATUS_THROTTLE)
+	else if("ircstatus" in input)	//tgs2 support
+		var/static/last_irc_status = 0
+		if(world.time - last_irc_status < 50)
```
Why did you hardcode this, not that it will be changed any time soon.

Cyberboss replied:
It's gonna get removed eventually, once the new tools are 100% reliable. I wanted to move the define to the __DEFINES tree
